clusterman/kubernetes/util.py

```diff
@@ -60,7 +60,18 @@ def __init__(self, kubeconfig_path: str, client_class: Type) -> None:
         :param Type client_class: k8s client class to initialize
         """
         try:
-            kubernetes.config.load_kube_config(kubeconfig_path, context=os.getenv("KUBECONTEXT"))
+            """
+            https://kubernetes.io/docs/concepts/containers/container-environment/#container-environment
+            Every pod in k8s get some default enviornment variable injected, Including KUBERNETES_SERVICE_HOST
+            which points to K8s service. We are using this variable to distinguise between when cluterman is started in
+            a pod vs when it's started on host. For clusterman services running inside a k8s cluster, We priiotise using
+            K8s Service account since that let us avoid creating any kubeconfig in advance.
+            For clusterman CLI invokation we continue using provided KUBECONFIG file
+            """
+            if os.getenv("KUBERNETES_SERVICE_HOST"):
+                kubernetes.config.load_incluster_config()
+            else:
+                kubernetes.config.load_kube_config(kubeconfig_path, context=os.getenv("KUBECONTEXT"))
```

Comment:
do we really set `KUBECONTEXT`? i thought the convention is to use `KUBECONFIG`. And imho that should be checked first to prefer overrides, otherwise assume we run in a k8s environment and can use the incluster config.

Reply:
I think it was added as a way to support `Kubeconfig` with multiple clusters. In clusterman this result into a empty string.
The reason to prefer using `KUBERNETES_SERVICE_HOST` is that, it's guaranteed to be injected by k8s inside a pod. So whenever this env is set we can assume that the clusterman inside a Pod.